index.js

```diff
@@ -186,11 +186,13 @@ module.exports = function (schema, options) {
         var versDoc = result;
         if (versDoc === null) {
           // Document doesn't exist so create a new one
+          delete dataObj.data._id;
           versDoc = new shadowModel(dataObj.data);
         } else {
           // Document does exist so copy data to it
           for (var key in dataObj.data) {
-            versDoc[key] = dataObj.data[key];
+            if(key !== '_id')
+              versDoc[key] = dataObj.data[key];
           }
         }
         versDoc.versionOfId = dataObj.versionOfId || null;
```

Comment:
Hello hepr,

your changes look all well imho.
I have a general question though regarding saveVersion() - I use upsertVersion() so I have never really come across this before.
Why should the dataObj contain a versionOfId field, isn't this one that gets inserted to the shadow collection only?
Wouldn't it be correct instead to say
- on line 189 instead of delete dataObj.data._id
  var dataCopy = dataObj.toObject(); delete dataCopy._id; versDoc = new shadowModel(dataCopy);
- and then on line 207 instead of model.findById(dataObj.versionOfId...)
  model.findById(dataObj._id...)
Also I think the field names are hard coded sometimes in saveVersion() where the 'Path' variables should be used:
- 182: dataObj.versionId should be dataObj[versionIdPath]
- 198: versDoc.versionOfId = dataObj.versionOfId... should be versDoc[versionOfIdPath] = dataObj[versionOfIdPath]... (if this line remains valid)
- 207: dataObj.versionOfId should be dataObj[versionOfIdPath] (if not replaced by dataObj._id)
- 215: same as 182

What do you think about this?
Regards,
Roman